src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryDataParser.java

```diff
@@ -63,49 +64,100 @@ private BigQueryDataParser() {
   }
 
   public static StructuredRecord getStructuredRecord(Schema schema, FieldList fields, FieldValueList fieldValues) {
+
     StructuredRecord.Builder recordBuilder = StructuredRecord.builder(schema);
-    for (Field field : fields) {
+    for (int i = 0; i < fields.size(); i++) {
```

Comment:
Is this change required? It just makes accessing the list of fields O(N^2) instead of O(N)